### PR TITLE
dev-util/memprof: Fix passing of incompatible pointer type

### PR DESCRIPTION
**Pull request by listout**
And update EAPI 7 -> 8

Closes: https://bugs.gentoo.org/923021

### CODE REVIEW

**Comment by listout** [comment]
Got a pkgdev warning
```
dev-util/memprof
  DeprecatedDep: version 0.6.2-r1: DEPEND: deprecated dependency: >=gnome-base/libglade-2
  DeprecatedDep: version 0.6.2-r1: RDEPEND: deprecated dependency: >=gnome-base/libglade-2
  DeprecatedDep: version 0.6.2-r2: DEPEND: deprecated dependency: >=gnome-base/libglade-2
  DeprecatedDep: version 0.6.2-r2: RDEPEND: deprecated dependency: >=gnome-base/libglade-2
```

It it worth fixing? or shall I close the bug?

File: dev-util/memprof/memprof-0.6.2-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+PATCHES=(
+	"${FILESDIR}"/${P}-binutils.patch
+	"${FILESDIR}"/${P}-desktop.patch
+	"${FILESDIR}"/${P}-gcc14-build-fix.patch
+)
+
+src_configure() {
+	econf \
+		--disable-static \
+		$(use_enable nls)
```
[the remark this answers is not in the document]
Comment (juippis) [reply dev-util/memprof/memprof-0.6.2-r2.ebuild:39]: @listout can you implement this? I guess there's not much to do about the libglade dep.

File: dev-util/memprof/memprof-0.6.2-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	nls? ( virtual/libintl )"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	virtual/pkgconfig
+	nls? (
+		dev-util/intltool
+		sys-devel/gettext
+	)"
+
+DOCS=( AUTHORS ChangeLog README NEWS )
```
[the remark this answers is not in the document]
Comment (juippis) [reply dev-util/memprof/memprof-0.6.2-r2.ebuild:28]: Yeah, agree.

[the remark this answers is not in the document]
Comment (listout) [reply dev-util/memprof/memprof-0.6.2-r2.ebuild]: Done, thanks to both

File: dev-util/memprof/memprof-0.6.2-r2.ebuild (hunk 1 above)
[the remark this answers is not in the document]
Comment (listout) [reply dev-util/memprof/memprof-0.6.2-r2.ebuild]: Done, thank you both.

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**